src/containers/EmployeeList/Employees/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { Box, ChevronIcon, Pagination, Select, TextInput } from '@mantine/core';
+import { useEffect, useRef, useState } from 'react';
+import Employee from '../Employee';
+import styles from './styles.module.scss';
+import IEmployee from '@/types/employee';
+
+interface employeesProps {
+  listEmployees: Array<IEmployee>;
+  isValidating: boolean;
+  pageNumber: number;
+}
+export default function Employees({
+  listEmployees,
+  isValidating,
+  pageNumber,
+}: employeesProps) {
+  const [employeeList, setEmployeeList] = useState(listEmployees);
+  const [activePage, setPage] = useState(1);
+  const [search, setSearch] = useState('');
+  const [gender, setGender] = useState('all');
+  const [arrange, setArrange] = useState('all');
+  const typingTimeOut = useRef<any>(null);
+  const ascending = (a: String, b: String) => {
+    if (a > b) return 1;
+    if (a < b) return -1;
+    return 0;
+  };
+  const descending = (a: String, b: String) => {
+    if (a < b) return 1;
+    if (a > b) return -1;
+    return 0;
+  };
+  const handleChangeSearch = (event: any) => {
+    if (typingTimeOut.current) {
+      clearTimeout(typingTimeOut.current);
+    }
+    typingTimeOut.current = setTimeout(() => {
+      setSearch(event.target.value);
+    }, 600);
+  };
+  useEffect(() => {
+    let filter = listEmployees.filter((value: IEmployee) => {
+      if (search === '') {
+        return value;
+      } else {
+        return value.name.toLowerCase().includes(search.toLowerCase().trim());
+      }
+    });
+    setEmployeeList(filter);
+  }, [search]);
+  console.log(arrange);
+  return (
+    <Box
+      sx={(theme) => ({
+        display: 'block',
+        backgroundColor:
+          theme.colorScheme === 'dark'
+            ? theme.colors.dark[4]
+            : theme.colors.gray[1],
+        color:
+          theme.colorScheme === 'dark'
+            ? theme.colors.blue[4]
+            : theme.colors.blue[7],
+        textAlign: 'center',
+        borderRadius: theme.radius.md,
+      })}
+      className={styles.mainBox}
+    >
+      <header className={styles.boxHeader}>
+        <div className={styles.containerBoxHeader}>
+          <TextInput
+            ref={typingTimeOut}
+            placeholder="Search"
+            className={styles.searchBox}
+            onChange={handleChangeSearch}
+          ></TextInput>
+          <Select
+            rightSection={<ChevronIcon />}
+            rightSectionWidth={30}
+            styles={{ rightSection: { pointerEvents: 'none' } }}
+            placeholder="All"
+            className={styles.selectionBox}
+            onChange={(event) => setArrange(event!)}
```

Comment:
@cpea2506 What do you think about this and similar onChange?

Reply:
If i remember right, we can get rid of callback and pass setArrange directly to `onChange` due to `Mantine Select onChange` props.

<img width="1006" alt="Screen Shot 2022-04-29 at 13 13 32" src="https://user-images.githubusercontent.com/42694704/165893204-7858b8db-5188-49ca-bf4e-572efb04ea4d.png">

Change to: 

```ts
onChange={setArrange}
```

Reply:
@sekke276 pls check this.